lib/pos_block_producer_test.go

```diff
@@ -33,9 +33,10 @@ func TestCreateBlockTemplate(t *testing.T) {
 	require.NoError(err)
 	dir := _dbDirSetup(t)
 
-	mempool := NewPosMempool(params, globalParams, latestBlockView, 2, dir, false,
-		maxMempoolPosSizeBytes, mempoolBackupIntervalMillis)
-	require.NoError(mempool.Init(1, nil, 1))
+	mempool := NewPosMempool()
+	require.NoError(mempool.Init(
```

Comment:
So just thinking about how we can future-proof this as we seem to have a ton of params for the Init function. Should we introduce a struct called `PoSMempoolInitParams` and validate that struct in `Init`? This way if we introduce new attributes in the future, we don't need to update a bunch of messy function sigs.

Reply:
Eh, passing a wrapper struct doesn't really simplify things that much. If we add a new param, we would still need to update the struct definition and update the callsites to populate the new struct field. It'll be roughly the same amount of code changes